proposed/event-dispatcher.md

```diff
@@ -66,54 +54,28 @@ A Listener MUST have a `void` return, and SHOULD type hint that return explicitl
-
-* MAY call relevant Listeners in any order it wishes.
-* MAY delay calling Listeners until some later point, such as using a queue system.
-* MAY call multiple Listeners concurrently (such as in an asynchronous system or a queue system with multiple worker processes).
-
-An Emitter calling a Notifier MUST NOT assume that any listeners have fired yet by the time the Notifier has returned.
-
-A Notifier MUST allow for the case where zero Listeners are found and MUST NOT generate an error condition in that case.
-
-### Error handling
+A Dispatcher is a service object implementing `EventDispatcherInterface`.  It is responsible for invoking listeners provided by a Listener Provider on an Event.
 
-A Notifier MUST ensure that any `\Throwable` generated by a listener does not impact or prevent other listeners from triggering normally.  It is RECOMMENDED that the Notifier log the error event but the only requirement is that one listener's exception MUST NOT block another listener.
-
-## Processor
-
-A Processor is a service object implementing `TaskProcessorInterface`.  It is used for cases where an Emitter wishes to provide data to listeners and receive data back from them.  Examples of Task use cases include:
-
-* Passing an object to a series of Listeners to allow it to be modified before it is saved to a persistence system.
-* Passing a collection to a series of Listeners to allow them to register values with it so that the Emitter may act on all of the collected information.
-* Passing a collection to a series of Listeners to allow them to modify the collection in some way before the Emitter takes action.
-* Passing some contextual information to a series of Listeners so that all of them may "vote" on what action to take, with the Emitter deciding based on the aggregate information provided.
-* Passing an object to a series of listeners and allowing one of them to set a value and then prevent further listeners from running.
-
-Tasks passed to a Processor SHOULD have some sort of mutator methods on them to allow Listeners to modify the object.  The nature of those methods is up to each implementation to determine.
-
-A Processor
+A Dispatcher
 
 * MUST call Listeners synchronously in the order they are returned from a ListenerProvider.
-* MUST return the Task it was passed after it is done invoking Listeners.
+* MUST return the same Event object it was passed after it is done invoking Listeners.
 * MUST NOT return to the Emitter until all Listeners have executed.
-* As an exception to the previous point, if the Task is a Promise then the the Dispatcher MAY return that Promise before all Listeners have executed.  However, the Promise MUST NOT be treated as fulfilled until all Listeners have executed.
+* As an exception to the previous point, if the Event is a Promise object then the the Dispatcher MAY return that Promise before all Listeners have executed.  However, the Promise MUST NOT be treated as fulfilled until all Listeners have executed.
```

Comment:
Why is this even in the spec? `Promise` is not in the list of definitions, and we've not spoken about anything async up to this point.

I tend to agree that this is how the system should operate in such a scenario, but I don't recall having this conversation at any point.

Reply:
It's something we discussed very early on, last spring.  I'm pretty sure it would be in the notes doc somewhere.  It was included in order to improve compatibility with async systems without addressing async directly.

Reply:
When I said, "we've not spoken about anything async up to this point," I'm talking about the specification itself. We definitely discussed it — but my recollection is we decided to just NOT address it AT ALL, as that allows implementers to decide whether or not they can handle async events/listeners/etc. and still comply with the spec.

My argument here is that it's a new addition to the spec, and my recollection is that as a WG, we decided it wasn't necessary to discuss it within the spec.

Reply:
It's not a new addition.  It's been in the spec since last spring, as evidenced by the fact that the line is being modified in place, not added. :smile: The only change in this line is s/Task/Event/ and adding the word "object" for linguistic clarity.

Revisting the existence of that line is out of scope for this PR, IMO.

Reply:
I saw the `+` and immediately thought "new addition". My bad!

Let's revisit after this PR is done.